lib/sparse/src/index/search_context.rs

```diff
@@ -900,70 +738,70 @@ mod tests {
         );
     }
 
-    #[test]
-    fn pruning_does_not_skip_negative_score_test() {
-        let inverted_index_ram = InvertedIndexBuilder::new()
-            .add(
-                1,
-                PostingList::from(vec![(1, 1.0), (2, 2.0), (3, 3.0), (4, 1.0), (5, -40.0)]),
-            )
-            .build();
-
-        let is_stopped = AtomicBool::new(false);
-        let mut search_context = SearchContext::new(
-            SparseVector {
-                indices: vec![1, 2, 3],
-                values: vec![-1.0, 1.0, 1.0],
-            },
-            2,
-            &inverted_index_ram,
-            &is_stopped,
-        );
-
-        // pruning is automatically deactivated because the query vector contains negative values
-        assert!(!search_context.use_pruning);
-        assert_eq!(
-            search_context.search(&match_all),
-            vec![
-                ScoredPointOffset {
-                    score: 40.0,
-                    idx: 5
-                },
-                ScoredPointOffset {
-                    score: -1.0,
-                    idx: 1
-                },
-            ]
-        );
-
-        // try again with pruning to show the problem
-        let mut search_context = SearchContext::new(
-            SparseVector {
-                indices: vec![1, 2, 3],
-                values: vec![-1.0, 1.0, 1.0],
-            },
-            2,
-            &inverted_index_ram,
-            &is_stopped,
-        );
-        search_context.use_pruning = true;
-        assert!(search_context.use_pruning);
-
-        // the last value has been pruned although it could have contributed a high score -1 * -40 = 40
-        assert_eq!(
-            search_context.search(&match_all),
-            vec![
-                ScoredPointOffset {
-                    score: -1.0,
-                    idx: 1
-                },
-                ScoredPointOffset {
-                    score: -2.0,
-                    idx: 2
-                }
-            ]
-        );
-    }
+    // #[test]
+    // fn pruning_does_not_skip_negative_score_test() {
```

Comment:
keeping it for the further work?

Reply:
Good catch :+1: 

Yes I want to revisit pruning later on and adapt this test.